### PR TITLE
Feature #50: Enable teams to configure additional columns for staging entities models

### PR DESCRIPTION
**Pull request by FrankTub**
**Please provide your name and company**

Frank Tubbing - Beequip

**Link the issue/feature request which this PR is meant to address**

https://github.com/fivetran/dbt_netsuite_source/issues/50

**Detail what changes this PR introduces and how this addresses the issue/feature request linked above.**

Enables me to define custom entity fields without having to create my own base/staging model.

**How did you validate the changes introduced within this PR?**

Ran my forked repo branch in my own dbt project

**Which warehouse did you use to develop these changes?**

Postgres

**Did you update the CHANGELOG?**

- [ ] Yes

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)**

- [ ] Yes

**Provide an emoji that best describes your current mood**
 
:dancer:

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

**PR Template** 
- [Community Pull Request Template](?expand=1&template=pull_request_template.md) (default)

- [Maintainer Pull Request Template](?expand=1&template=maintainer_pull_request_template.md) (to be used by maintainers)

### CODE REVIEW

**Comment by FrankTub** [comment]
Not really sure if I should bump the version here ?

**Comment by fivetran-avinash** [comment]
> Not really sure if I should bump the version here ?

Hi @FrankTub ! Yes, you'll want to bump the version up. Since I don't think any customers would be impacted, I'd recommend a minor update up to version 0.10.1.

**Comment by fivetran-avinash** [comment]
Thanks for opening this PR @FrankTub ! I think this mostly looks good--I've tested these changes and I feel mostly confident these are correct. 

I was wondering if you had any issue with me making some commits directly to your branch? There are a few changes within our `integration_tests` folder and other general maintenance changes we include for every release. 

If it works for you, as I review your changes I can make these background changes to your branch. This way we may fold them into the `main` branch more smoothly for a release early next week.

If not, I can provide you with the required changes as well in my PR review.

Once you give me the approval either way, we can kickstart the process of this release! Let me know your questions and concerns.

**Comment by fivetran-avinash** [comment]
@FrankTub Also one additional question: Is `group` one of the columns you are attempting to passthrough? There are some intricacies we will need to address on our end that we will add to the release if that's the case, as `group` is a reserved keyword and our current passthrough column logic isn't quite equipped to handle this. But we have workarounds in that case.

**Comment by FrankTub** [comment]
@fivetran-avinash ,  the field `group` is not per se one of the fields we will need to pass through. 

You are free to add commits to my branch without breaking any functionality at our side. We want to start using this, but don't use it yet in our production environment. I've bumped the version in my branch to `0.10.1`.

File: macros/get_entity_columns.sql
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply macros/get_entity_columns.sql:1]: Removed these staging columns.

**Review by fivetran-avinash — APPROVED** [review]
Thanks @FrankTub for letting us know about this feature and then opening this PR proactively! We really appreciate your collaboration and patience as we get this to a ready state. 

I made a few of the changes I discussed to ensure this passthrough column feature works properly on the entity staging macro and model. 

With that done, I've approved this PR and have begun kicking off the internal release process. Once this is release reviewed, I will merge this into main, and you should see this new release live on our hub hopefully sometime this week! 

We will keep you in the loop.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+# dbt_netsuite_source v0.10.1
+[PR #51](https://github.com/fivetran/dbt_netsuite_source/pull/51) includes the following update:
+
+## Feature Update
+- Added the `entities_pass_through_columns` variable to be leveraged within the `stg_netsuite__entities` model. This variable will allow users to pass through fields which are not natively included to the package staging model.
+  - Removed fields from the `get_entity_columns` macro that are not brought into the `stg_netsuite__entities` model to ensure that the passthrough feature works properly.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:6]: The entities model is only available in Nestuite2 and the model is netsuite2 named. Requested suggestion to clarify this.
```suggestion
- Added the `entities_pass_through_columns` variable to be leveraged within the `stg_netsuite2__entities` model. This variable will allow users to pass through fields which are not natively included to the package staging model.
  - Removed fields from the `get_entity_columns` macro that are not brought into the `stg_netsuite2__entities` model to ensure that the passthrough feature works properly.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+ 
+## Contributors:
+- [@FrankTub](https://github.com/FrankTub)
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:9]: While it is a bit repetitive, I would like to include the PR link next to the contributor name. This way it is easy to see where their contributions may be found.
```suggestion
- [@FrankTub](https://github.com/FrankTub)  ([PR #51](https://github.com/fivetran/dbt_netsuite_source/pull/51))
```

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@FrankTub @fivetran-avinash thanks for working through this feature update! This PR looks great, I just have two small suggestions to be applied before approval.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM

**Pull request merged**